### PR TITLE
Rewrite the enchanter addon

### PR DESCRIPTION
**Pull request by keneanung**
This improves the enchanter into different directions:
 - Allow for enchantments of full orders with one command similar to the
   refiller
 - Allow the enchanter to tell the user the commodity cost of an order
 - Speed enchantment process up a bit by not using "doadd" for balancefree
   ending actions

### CODE REVIEW

File: doc/enchanter.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  vconfig haveEnchantmentMedaillon
+    Toggles whether you have the Medaillon of Enchantment or not.
+
+
+API
+^^^
+.. glossary::
+
+  svo.getEnchantCost(what, howMany, costCollection)
```
Comment (vadi2) [inline doc/enchanter.rst]: API functions in Svof generally are lowercase, could this be updated?

File: doc/enchanter.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  enchant cancel
+    Stops the current order after the currently running enchantment.
+
+  enchant cancel force
+    Immediately stops the current enchantment order
+
+  reenchant
+    Repeats the last order.
+
+  vconfig enchantGetGold <command>
```
Comment (vadi2) [inline doc/enchanter.rst]: vconfig params in Svof generally are lowercase, could this be updated?

**Comment by keneanung** [comment]
Done. Also added the rules to the coding style wiki page.

**Comment by vadi2** [comment]
Appreciated. I missed those two rules.

**Pull request merged**